### PR TITLE
Manifest files use env variable to set author or default back to the …

### LINKED ISSUE #1484: The manifest files should set the `author` key from an environment variable for dev purposes otherwise default back to the `realname()`
When creating a new project with `forc init <name>` the created manifest files set the author to be the `realname()` i.e. the user.
This is desirable behaviour for external use however for development purposes we would like to have the author set to "Fuel Labs \<contact@fuel.sh>".

There should be an environment variable that is checked and used if it is set otherwise default back to the `realname()` for the author.

### PR DESCRIPTION
**Pull request by Braqzen**
…realname

Closes #1484

### CODE REVIEW

File: forc/src/utils/defaults.rs
Comment (adlerjohn) [inline forc/src/utils/defaults.rs:4]: Abstract this string into a constant since it's used more than once and is part of the public API.

Comment (adlerjohn) [reply forc/src/utils/defaults.rs]: Fixed in [afe7cd6](https://github.com/FuelLabs/sway/pull/1487/commits/afe7cd62ff64983632cf2c9bf4b17be7074c47ad)

**Review by adlerjohn — DISMISSED** [review]
Looks fine to me, but will defer to tooling team.

**Review by eureka-cpu — COMMENTED** [review]
If possible, since we're using the same code twice here, could you make a function possibly called `get_author` and call it for those variables? Just trying to tackle code bloat when I see it so we don't have to go back and fix it later, if this is a blocker it could also just be added to #1107 for a separate PR

File: forc/src/utils/defaults.rs
Comment (otrho) [inline forc/src/utils/defaults.rs:7]: FYI a simpler way to do this pattern is:
```rust
let author =
    std::env::var(sway_utils::FORC_INIT_MANIFEST_AUTHOR).unwrap_or_else(|| whoami::realname());
```

**Review by otrho — DISMISSED** [review]

**Review by JoshuaBatty — CHANGES_REQUESTED** [review]
Thanks @Braqzen! Can you please make the suggested edits that were made by @otrho and @eureka-cpu first before landing this?

**Review by eureka-cpu — APPROVED** [review]
looking clean, thanks!

**Review by JoshuaBatty — APPROVED** [review]

**Comment by JoshuaBatty** [comment]
Looks good, thanks @Braqzen

**Pull request merged**
